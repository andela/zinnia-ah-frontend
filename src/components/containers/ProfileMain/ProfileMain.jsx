/* eslint-disable no-console */
/* eslint-disable react/prop-types */
/* eslint-disable react/display-name */
import React from 'react';
import { Tab, Icon } from 'semantic-ui-react';

import Title from '../../presentationals/Title/Title.jsx';
import ProfileSettings from '../ProfileSettings/ProfileSettings.jsx';
import './ProfileMain.scss';

export default function ProfileMain(props) {
  const { firstName, lastName, bio, publications } = props.profile;
  const panes = [
    {
      menuItem: 'Posts',
      render: () => <Tab.Pane attached={false}>{publications}</Tab.Pane>,
    },
    {
      menuItem: 'Stats',
      render: () => <Tab.Pane attached={false}>Tab 2 Content</Tab.Pane>,
    },
    {
      menuItem: 'Likes',
      render: () => <Tab.Pane attached={false}>Tab 3 Content</Tab.Pane>,
    },
    {
      menuItem: 'Comments',
      render: () => <Tab.Pane attached={false}>Tab 4 Content</Tab.Pane>,
    },
    {
      menuItem: 'Bookmarks',
      render: () => <Tab.Pane attached={false}>Tab 5 Content</Tab.Pane>,
    },
    {
      menuItem: 'Reports',
      render: () => <Tab.Pane attached={false}>Tab 6 Content</Tab.Pane>,
    },
    {
      menuItem: 'Settings',
      render: () => (
        <Tab.Pane attached={false}>
          <ProfileSettings />
        </Tab.Pane>
      ),
    },
  ];

  return (
    <div className="main">
      {!firstName || !lastName ? (
        <div
          style={{
            display: 'flex',
          }}
        >
          <Title content="Full Name" className="title-lg fade-2" />
          <Icon
            name="edit outline"
            style={{
              fontSize: '20px',
              marginLeft: '40px',
            }}
          />
        </div>
      ) : (
        <Title content={`${firstName} ${lastName}`} className="title-lg" />
      )}

      <div
        style={{
          maxHeight: '200px',
        }}
      >
        {bio ? (
          <p
            style={{
              fontWeight: 'lighter',
              fontSize: '18px',
              lineHeight: '27px',
              marginBottom: '20px',
            }}
          >
            {bio}
          </p>
        ) : (
          <p
            style={{
              fontWeight: 'lighter',
              fontSize: '18px',
              lineHeight: '27px',
              marginBottom: '20px',
            }}
          >
            Please tell us about yourself by editing your profile
          </p>
        )}
      </div>
      <div className="profile-tab-menu">
        <Tab menu={{ secondary: true, pointing: true }} panes={panes} />
      </div>
    </div>
  );
}
