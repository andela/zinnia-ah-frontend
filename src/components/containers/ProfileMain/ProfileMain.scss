@import url('https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento');
@import '../../../assets/variables';

.main {
  font-family: 'Lato', sans-serif;
  padding-left: 2%;
  width: 75%;

  .name-title {
    font-size: 30px;
    line-height: 38px;
    margin-bottom: 20px;
  }
}

.post-lists {

  .item {
    margin-top: 30px !important;

    .content {
      display: flex !important;
      flex-direction: column;
    }

    .image {
      height: 210px;
      width: 210px !important;

      img {
        height: 210px !important;
        width: 100%;
      }
    }
  }
}

.fade-2 {
  opacity: .2;
}

.profile-tab-menu {

  .menu {
    border-bottom: 2px solid $secondary-grey;
    display: flex;
    margin-bottom: 0;
    width: 100%;

    .item {
      color: $secondary-black;
      font-family: 'Lato', sans-serif;
      font-size: 24px;
      line-height: 29px;
    }

    .active.item {
      font-weight: 400 !important;
    }
  }

  .segment {
    border: 0;
    border-radius: 0;
    box-shadow: none;
    margin: 0;
    padding: 0;
  }
}

.settings {
  margin-top: 30px;
}

.edit-profile {
  background: $primary-white;
  border: 1px solid $secondary-grey;
  border-radius: 4px;
  box-sizing: border-box;
  padding: 75px 21px 24px;
}

@media only screen and (max-width: 600px) {

  .main {
    width: 100%;
  }
}
