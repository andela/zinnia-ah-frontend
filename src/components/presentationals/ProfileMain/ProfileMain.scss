@import '../../../assets/variables';

.main {
  font-family: $lato;
  padding-left: 2%;
  width: 75%;

  .name-title {
    font-size: 1.875rem;
    line-height: 2.375rem;
    margin-bottom: 1.25rem;
  }
}

.post-lists {
  .item {
    margin-top: 1.875rem !important;

    .content {
      display: flex !important;
      flex-direction: column;
    }

    .image {
      height: 13.125rem;
      width: 13.125rem !important;

      img {
        height: 13.125rem !important;
        width: 100%;
      }
    }
  }
}

.fade-2 {
  opacity: .2;
}

.profile-tab-menu {
  .menu {
    border-bottom: .125rem solid $secondary-grey;
    display: flex;
    margin-bottom: 0;
    width: 100%;

    .item {
      color: $secondary-black;
      font-family: $lato;
      font-size: 1.5rem;
      line-height: 1.8125rem;
    }

    .active.item {
      font-weight: 400 !important;
    }
  }

  .segment {
    border: 0;
    border-radius: 0;
    box-shadow: none;
    margin: 0;
    padding: 0;
  }
}

.settings {
  margin-top: 1.875rem;
}

.edit-profile {
  background: $primary-white;
  border: .0625rem solid $secondary-grey;
  border-radius: .25rem;
  box-sizing: border-box;
  padding: 4.6875rem 1.3125rem 1.5rem;
}

@media only screen and (max-width: 37.5rem) {

  .main {
    width: 100%;
  }
}
