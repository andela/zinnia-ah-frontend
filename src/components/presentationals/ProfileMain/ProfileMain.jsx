/* eslint-disable react/display-name */
import React from 'react';
import { Tab } from 'semantic-ui-react';
import PropTypes from 'prop-types';

// components
import Title from '../Title/Title';
import ArticleLists from '../ArticleLists/ArticleLists';
import ProfileSettings from '../ProfileSettings/ProfileSettings';

// styles
import './ProfileMain.scss';

const ProfileMain = ({
  firstName,
  lastName,
  bio,
  publications,
  deleteArticle,
  isDeleting,
}) => {
  const panes = [
    {
      menuItem: 'Posts',
      render: () => (
        <Tab.Pane attached={false}>
          <ArticleLists
            articles={publications}
            deleteArticle={deleteArticle}
            isDeleting={isDeleting}
          />
        </Tab.Pane>
      ),
    },
    {
      menuItem: 'Stats',
      render: () => <Tab.Pane attached={false}>Tab 2 Content</Tab.Pane>,
    },
    {
      menuItem: 'Likes',
      render: () => <Tab.Pane attached={false}>Tab 3 Content</Tab.Pane>,
    },
    {
      menuItem: 'Comments',
      render: () => <Tab.Pane attached={false}>Tab 4 Content</Tab.Pane>,
    },
    {
      menuItem: 'Bookmarks',
      render: () => <Tab.Pane attached={false}>Tab 5 Content</Tab.Pane>,
    },
    {
      menuItem: 'Reports',
      render: () => <Tab.Pane attached={false}>Tab 6 Content</Tab.Pane>,
    },
    {
      menuItem: 'Settings',
      render: () => (
        <Tab.Pane attached={false}>
          <ProfileSettings />
        </Tab.Pane>
      ),
    },
  ];

  return (
    <div className="main">
      <Title content={`${firstName} ${lastName}`} className="title-lg" />
      <div
        style={{
          maxHeight: '200px',
        }}
      >
        {bio}
      </div>
      <div className="profile-tab-menu">
        <Tab menu={{ secondary: true, pointing: true }} panes={panes} />
      </div>
    </div>
  );
};

ProfileMain.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  bio: PropTypes.string,
  publications: PropTypes.array,
  deleteArticle: PropTypes.func,
  isDeleting: PropTypes.bool,
};

export default ProfileMain;
