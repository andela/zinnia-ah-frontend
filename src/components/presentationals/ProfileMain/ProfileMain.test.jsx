import React from 'react';
import { mount } from 'enzyme';

import ProfileMain from './ProfileMain';

const props = {
  firstName: 'mr',
  lastName: 'user',
  bio: 'I am the correct user',
  publications: [],
  followings: [],
  followers: [],
  bookmarks: [],
  image: 'http://dummyImage.jpg',
  email: 'dummy@user.com',
  username: 'dummyuser',
  error: {},
  isLoading: true,
  isDeleting: false,
};

const undefinedProps = {
  publications: [],
};

const lastNameUndefinedProps = {
  firstName: 'ererger',
  bio: 'egrerg',
  publications: [],
};
describe('ProfileMain', () => {
  it('should render correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should not render correctly', () => {
    const component = mount(<ProfileMain {...lastNameUndefinedProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should not render correctly', () => {
    const component = mount(<ProfileMain {...undefinedProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should render Posts Tab correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render Stats Tab correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    const stats = component.find('MenuItem[name="Stats"]').simulate('click');
    expect(stats).toMatchSnapshot();
  });

  it('should render Likes Tab correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    const likes = component.find('MenuItem[name="Likes"]').simulate('click');
    expect(likes).toMatchSnapshot();
  });

  it('should render Comments Tab correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    const comments = component
      .find('MenuItem[name="Comments"]')
      .simulate('click');
    expect(comments).toMatchSnapshot();
  });

  it('should render Reports Tab correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    const reports = component
      .find('MenuItem[name="Reports"]')
      .simulate('click');
    expect(reports).toMatchSnapshot();
  });

  it('should render Settings Tab correctly', () => {
    const component = mount(<ProfileMain {...props} />);
    expect(component).toMatchSnapshot();
  });
});
