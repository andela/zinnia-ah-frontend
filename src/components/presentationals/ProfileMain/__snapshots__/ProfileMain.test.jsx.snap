// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMain should not render correctly 1`] = `
<ProfileMain
  bio="egrerg"
  firstName="ererger"
  publications={Array []}
>
  <div
    className="main"
  >
    <Title
      className="title-lg"
      content="ererger undefined"
    >
      <h1
        className="title title-lg"
      >
        ererger undefined
      </h1>
    </Title>
    <div
      style={
        Object {
          "maxHeight": "200px",
        }
      }
    >
      egrerg
    </div>
    <div
      className="profile-tab-menu"
    >
      <Tab
        grid={
          Object {
            "paneWidth": 12,
            "tabWidth": 4,
          }
        }
        menu={
          Object {
            "pointing": true,
            "secondary": true,
          }
        }
        panes={
          Array [
            Object {
              "menuItem": "Posts",
              "render": [Function],
            },
            Object {
              "menuItem": "Stats",
              "render": [Function],
            },
            Object {
              "menuItem": "Likes",
              "render": [Function],
            },
            Object {
              "menuItem": "Comments",
              "render": [Function],
            },
            Object {
              "menuItem": "Bookmarks",
              "render": [Function],
            },
            Object {
              "menuItem": "Reports",
              "render": [Function],
            },
            Object {
              "menuItem": "Settings",
              "render": [Function],
            },
          ]
        }
        renderActiveOnly={true}
      >
        <div>
          <Menu
            activeIndex={0}
            items={
              Array [
                "Posts",
                "Stats",
                "Likes",
                "Comments",
                "Bookmarks",
                "Reports",
                "Settings",
              ]
            }
            onItemClick={[Function]}
            pointing={true}
            secondary={true}
          >
            <div
              className="ui pointing secondary menu"
            >
              <MenuItem
                active={true}
                content="Posts"
                index={0}
                key="Posts"
                name="Posts"
                onClick={[Function]}
              >
                <a
                  className="active item"
                  onClick={[Function]}
                >
                  Posts
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Stats"
                index={1}
                key="Stats"
                name="Stats"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Stats
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Likes"
                index={2}
                key="Likes"
                name="Likes"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Likes
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Comments"
                index={3}
                key="Comments"
                name="Comments"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Comments
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Bookmarks"
                index={4}
                key="Bookmarks"
                name="Bookmarks"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Bookmarks
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Reports"
                index={5}
                key="Reports"
                name="Reports"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Reports
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Settings"
                index={6}
                key="Settings"
                name="Settings"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Settings
                </a>
              </MenuItem>
            </div>
          </Menu>
          <TabPane
            active={true}
            as={[Function]}
            attached={false}
          >
            <Segment
              attached={false}
              className="active tab"
            >
              <div
                className="ui segment active tab"
              >
                <ArticleLists
                  articles={Array []}
                >
                  <div
                    className="post-lists"
                  >
                    <p>
                      You have no publications
                    </p>
                  </div>
                </ArticleLists>
              </div>
            </Segment>
          </TabPane>
        </div>
      </Tab>
    </div>
  </div>
</ProfileMain>
`;

exports[`ProfileMain should not render correctly 2`] = `
<ProfileMain
  publications={Array []}
>
  <div
    className="main"
  >
    <Title
      className="title-lg"
      content="undefined undefined"
    >
      <h1
        className="title title-lg"
      >
        undefined undefined
      </h1>
    </Title>
    <div
      style={
        Object {
          "maxHeight": "200px",
        }
      }
    />
    <div
      className="profile-tab-menu"
    >
      <Tab
        grid={
          Object {
            "paneWidth": 12,
            "tabWidth": 4,
          }
        }
        menu={
          Object {
            "pointing": true,
            "secondary": true,
          }
        }
        panes={
          Array [
            Object {
              "menuItem": "Posts",
              "render": [Function],
            },
            Object {
              "menuItem": "Stats",
              "render": [Function],
            },
            Object {
              "menuItem": "Likes",
              "render": [Function],
            },
            Object {
              "menuItem": "Comments",
              "render": [Function],
            },
            Object {
              "menuItem": "Bookmarks",
              "render": [Function],
            },
            Object {
              "menuItem": "Reports",
              "render": [Function],
            },
            Object {
              "menuItem": "Settings",
              "render": [Function],
            },
          ]
        }
        renderActiveOnly={true}
      >
        <div>
          <Menu
            activeIndex={0}
            items={
              Array [
                "Posts",
                "Stats",
                "Likes",
                "Comments",
                "Bookmarks",
                "Reports",
                "Settings",
              ]
            }
            onItemClick={[Function]}
            pointing={true}
            secondary={true}
          >
            <div
              className="ui pointing secondary menu"
            >
              <MenuItem
                active={true}
                content="Posts"
                index={0}
                key="Posts"
                name="Posts"
                onClick={[Function]}
              >
                <a
                  className="active item"
                  onClick={[Function]}
                >
                  Posts
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Stats"
                index={1}
                key="Stats"
                name="Stats"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Stats
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Likes"
                index={2}
                key="Likes"
                name="Likes"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Likes
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Comments"
                index={3}
                key="Comments"
                name="Comments"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Comments
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Bookmarks"
                index={4}
                key="Bookmarks"
                name="Bookmarks"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Bookmarks
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Reports"
                index={5}
                key="Reports"
                name="Reports"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Reports
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Settings"
                index={6}
                key="Settings"
                name="Settings"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Settings
                </a>
              </MenuItem>
            </div>
          </Menu>
          <TabPane
            active={true}
            as={[Function]}
            attached={false}
          >
            <Segment
              attached={false}
              className="active tab"
            >
              <div
                className="ui segment active tab"
              >
                <ArticleLists
                  articles={Array []}
                >
                  <div
                    className="post-lists"
                  >
                    <p>
                      You have no publications
                    </p>
                  </div>
                </ArticleLists>
              </div>
            </Segment>
          </TabPane>
        </div>
      </Tab>
    </div>
  </div>
</ProfileMain>
`;

exports[`ProfileMain should render Bookmarks Tab correctly 1`] = `
<MenuItem
  active={false}
  content="Bookmarks"
  index={4}
  key="Bookmarks"
  name="Bookmarks"
  onClick={[Function]}
>
  <a
    className="item"
    onClick={[Function]}
  >
    Bookmarks
  </a>
</MenuItem>
`;

exports[`ProfileMain should render Comments Tab correctly 1`] = `
<MenuItem
  active={false}
  content="Comments"
  index={3}
  key="Comments"
  name="Comments"
  onClick={[Function]}
>
  <a
    className="item"
    onClick={[Function]}
  >
    Comments
  </a>
</MenuItem>
`;

exports[`ProfileMain should render Likes Tab correctly 1`] = `
<MenuItem
  active={false}
  content="Likes"
  index={2}
  key="Likes"
  name="Likes"
  onClick={[Function]}
>
  <a
    className="item"
    onClick={[Function]}
  >
    Likes
  </a>
</MenuItem>
`;

exports[`ProfileMain should render Posts Tab correctly 1`] = `
<ProfileMain
  bio="I am the correct user"
  email="dummy@user.com"
  error={Object {}}
  firstName="mr"
  followers={Array []}
  followings={Array []}
  image="http://dummyImage.jpg"
  isDeleting={false}
  isLoading={true}
  lastName="user"
  publications={Array []}
  username="dummyuser"
>
  <div
    className="main"
  >
    <Title
      className="title-lg"
      content="mr user"
    >
      <h1
        className="title title-lg"
      >
        mr user
      </h1>
    </Title>
    <div
      style={
        Object {
          "maxHeight": "200px",
        }
      }
    >
      I am the correct user
    </div>
    <div
      className="profile-tab-menu"
    >
      <Tab
        grid={
          Object {
            "paneWidth": 12,
            "tabWidth": 4,
          }
        }
        menu={
          Object {
            "pointing": true,
            "secondary": true,
          }
        }
        panes={
          Array [
            Object {
              "menuItem": "Posts",
              "render": [Function],
            },
            Object {
              "menuItem": "Stats",
              "render": [Function],
            },
            Object {
              "menuItem": "Likes",
              "render": [Function],
            },
            Object {
              "menuItem": "Comments",
              "render": [Function],
            },
            Object {
              "menuItem": "Bookmarks",
              "render": [Function],
            },
            Object {
              "menuItem": "Reports",
              "render": [Function],
            },
            Object {
              "menuItem": "Settings",
              "render": [Function],
            },
          ]
        }
        renderActiveOnly={true}
      >
        <div>
          <Menu
            activeIndex={0}
            items={
              Array [
                "Posts",
                "Stats",
                "Likes",
                "Comments",
                "Bookmarks",
                "Reports",
                "Settings",
              ]
            }
            onItemClick={[Function]}
            pointing={true}
            secondary={true}
          >
            <div
              className="ui pointing secondary menu"
            >
              <MenuItem
                active={true}
                content="Posts"
                index={0}
                key="Posts"
                name="Posts"
                onClick={[Function]}
              >
                <a
                  className="active item"
                  onClick={[Function]}
                >
                  Posts
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Stats"
                index={1}
                key="Stats"
                name="Stats"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Stats
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Likes"
                index={2}
                key="Likes"
                name="Likes"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Likes
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Comments"
                index={3}
                key="Comments"
                name="Comments"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Comments
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Bookmarks"
                index={4}
                key="Bookmarks"
                name="Bookmarks"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Bookmarks
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Reports"
                index={5}
                key="Reports"
                name="Reports"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Reports
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Settings"
                index={6}
                key="Settings"
                name="Settings"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Settings
                </a>
              </MenuItem>
            </div>
          </Menu>
          <TabPane
            active={true}
            as={[Function]}
            attached={false}
          >
            <Segment
              attached={false}
              className="active tab"
            >
              <div
                className="ui segment active tab"
              >
                <ArticleLists
                  articles={Array []}
                  isDeleting={false}
                >
                  <div
                    className="post-lists"
                  >
                    <p>
                      You have no publications
                    </p>
                  </div>
                </ArticleLists>
              </div>
            </Segment>
          </TabPane>
        </div>
      </Tab>
    </div>
  </div>
</ProfileMain>
`;

exports[`ProfileMain should render Reports Tab correctly 1`] = `
<MenuItem
  active={false}
  content="Reports"
  index={5}
  key="Reports"
  name="Reports"
  onClick={[Function]}
>
  <a
    className="item"
    onClick={[Function]}
  >
    Reports
  </a>
</MenuItem>
`;

exports[`ProfileMain should render Settings Tab correctly 1`] = `
<MenuItem
  active={false}
  content="Settings"
  index={6}
  key="Settings"
  name="Settings"
  onClick={[Function]}
>
  <a
    className="item"
    onClick={[Function]}
  >
    Settings
  </a>
</MenuItem>
`;

exports[`ProfileMain should render Stats Tab correctly 1`] = `
<MenuItem
  active={false}
  content="Stats"
  index={1}
  key="Stats"
  name="Stats"
  onClick={[Function]}
>
  <a
    className="item"
    onClick={[Function]}
  >
    Stats
  </a>
</MenuItem>
`;

exports[`ProfileMain should render correctly 1`] = `
<ProfileMain
  bio="I am the correct user"
  email="dummy@user.com"
  error={Object {}}
  firstName="mr"
  followers={Array []}
  followings={Array []}
  image="http://dummyImage.jpg"
  isDeleting={false}
  isLoading={true}
  lastName="user"
  publications={Array []}
  username="dummyuser"
>
  <div
    className="main"
  >
    <Title
      className="title-lg"
      content="mr user"
    >
      <h1
        className="title title-lg"
      >
        mr user
      </h1>
    </Title>
    <div
      style={
        Object {
          "maxHeight": "200px",
        }
      }
    >
      I am the correct user
    </div>
    <div
      className="profile-tab-menu"
    >
      <Tab
        grid={
          Object {
            "paneWidth": 12,
            "tabWidth": 4,
          }
        }
        menu={
          Object {
            "pointing": true,
            "secondary": true,
          }
        }
        panes={
          Array [
            Object {
              "menuItem": "Posts",
              "render": [Function],
            },
            Object {
              "menuItem": "Stats",
              "render": [Function],
            },
            Object {
              "menuItem": "Likes",
              "render": [Function],
            },
            Object {
              "menuItem": "Comments",
              "render": [Function],
            },
            Object {
              "menuItem": "Bookmarks",
              "render": [Function],
            },
            Object {
              "menuItem": "Reports",
              "render": [Function],
            },
            Object {
              "menuItem": "Settings",
              "render": [Function],
            },
          ]
        }
        renderActiveOnly={true}
      >
        <div>
          <Menu
            activeIndex={0}
            items={
              Array [
                "Posts",
                "Stats",
                "Likes",
                "Comments",
                "Bookmarks",
                "Reports",
                "Settings",
              ]
            }
            onItemClick={[Function]}
            pointing={true}
            secondary={true}
          >
            <div
              className="ui pointing secondary menu"
            >
              <MenuItem
                active={true}
                content="Posts"
                index={0}
                key="Posts"
                name="Posts"
                onClick={[Function]}
              >
                <a
                  className="active item"
                  onClick={[Function]}
                >
                  Posts
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Stats"
                index={1}
                key="Stats"
                name="Stats"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Stats
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Likes"
                index={2}
                key="Likes"
                name="Likes"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Likes
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Comments"
                index={3}
                key="Comments"
                name="Comments"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Comments
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Bookmarks"
                index={4}
                key="Bookmarks"
                name="Bookmarks"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Bookmarks
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Reports"
                index={5}
                key="Reports"
                name="Reports"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Reports
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="Settings"
                index={6}
                key="Settings"
                name="Settings"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  Settings
                </a>
              </MenuItem>
            </div>
          </Menu>
          <TabPane
            active={true}
            as={[Function]}
            attached={false}
          >
            <Segment
              attached={false}
              className="active tab"
            >
              <div
                className="ui segment active tab"
              >
                <ArticleLists
                  articles={Array []}
                  isDeleting={false}
                >
                  <div
                    className="post-lists"
                  >
                    <p>
                      You have no publications
                    </p>
                  </div>
                </ArticleLists>
              </div>
            </Segment>
          </TabPane>
        </div>
      </Tab>
    </div>
  </div>
</ProfileMain>
`;
